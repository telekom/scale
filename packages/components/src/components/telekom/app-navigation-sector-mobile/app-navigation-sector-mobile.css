/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

app-navigation-sector-mobile {
  --border-bottom: 1px solid var(--telekom-color-ui-additional-subtle);
  --color: var(--telekom-color-text-and-icon-standard);
  --font-weight: var(--telekom-typography-font-weight-bold);
  --font-size: var(--telekom-typography-font-size-body);
  --transition: all var(--telekom-motion-duration-transition)
    cubic-bezier(var(--telekom-motion-easing-standard));
  --border-bottom-selected: 1px solid var(--telekom-color-primary-standard);
  --color-selected: var(--telekom-color-text-and-icon-primary-standard);
  width: 100%;
}

.sector-navigation-mobile {
  display: flex;
  width: 100%;
  list-style: none;
  padding: 0;
  margin: 0;
  border-bottom: var(--border-bottom);
}

.sector-navigation-mobile__item {
  width: 100%;
}

.sector-navigation-mobile__item-link {
  color: var(--color);
  text-decoration: none;
  font-weight: var(--font-weight);
  font-size: var(--font-size);
  width: 100%;
  height: 54px;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: var(--transition);
}

.sector-navigation-mobile__item-link--selected {
  border-bottom: var(--border-bottom-selected);
  color: var(--color-selected);
  transition: var(--transition);
}
