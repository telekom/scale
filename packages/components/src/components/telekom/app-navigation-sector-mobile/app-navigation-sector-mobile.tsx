/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { Component, h, Prop, State, Watch } from '@stencil/core';
import { MenuItem } from '../app-interfaces';

@Component({
  tag: 'app-navigation-sector-mobile',
  styleUrl: 'app-navigation-sector-mobile.css',
})
export class NavigationSectorMobile {
  @Prop() hide: () => void;
  @Prop() navigation: MenuItem[];
  @Prop() activeSectorId?: string;
  // @ts-ignore
  @State() selected: MenuItem = this.navigation
    ? // @ts-ignore
      this.navigation.find(({ id }) => id === this.activeSectorId) ||
      // @ts-ignore
      this.navigation[0]
    : ({} as MenuItem);

  @Watch('activeSectorId')
  handleActiveSegment(newValue) {
    this.selected =
      this.navigation.find(({ id }) => id === newValue) || this.navigation[0];
  }

  handleSelected(event, item) {
    this.selected = item;
    if (typeof item.onClick === 'function') {
      item.onClick(event);
    }
  }

  render() {
    return (
      <ul class="sector-navigation-mobile">
        {(this.navigation || []).map((item) => (
          <li class="sector-navigation-mobile__item">
            <a
              class={`sector-navigation-mobile__item-link${
                this.selected.id === item.id
                  ? ' sector-navigation-mobile__item-link--selected'
                  : ''
              }`}
              href={item.href || 'javascript:void(0);'}
              onClick={(event) => this.handleSelected(event, item)}
              onKeyDown={(event) => {
                if (['Escape', 'Esc'].includes(event.key)) {
                  this.hide();
                }
              }}
              aria-current={this.selected.id === item.id ? 'true' : 'false'}
            >
              {item.name}
              {this.selected.id === item.id && (
                <span class="sr-only">active</span>
              )}
            </a>
          </li>
        ))}
      </ul>
    );
  }
}
