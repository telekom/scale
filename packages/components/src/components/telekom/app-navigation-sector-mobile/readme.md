# app-navigation-sector-mobile



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute          | Description | Type         | Default     |
| ---------------- | ------------------ | ----------- | ------------ | ----------- |
| `activeSectorId` | `active-sector-id` |             | `string`     | `undefined` |
| `hide`           | `hide`             |             | `() => void` | `undefined` |
| `navigation`     | `navigation`       |             | `MenuItem[]` | `undefined` |


## Dependencies

### Used by

 - [scale-app-header](../app-header)

### Graph
```mermaid
graph TD;
  scale-app-header --> app-navigation-sector-mobile
  style app-navigation-sector-mobile fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
