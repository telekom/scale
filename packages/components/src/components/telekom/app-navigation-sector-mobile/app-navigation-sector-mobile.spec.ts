/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newSpecPage } from '@stencil/core/testing';
import { NavigationSectorMobile } from './app-navigation-sector-mobile';
import { MenuItem } from '../app-interfaces';

const nav = {
  name: 'nav1',
  id: 'menuItem1',
  href: '#',
  children: [
    {
      name: 'menuItem1-1',
      id: 'menuItem1-1',
      href: '#',
      level: 2,
      onClick: (event: KeyboardEvent) => {
        return event.key;
      },
    },
  ],
  level: 1,
};

describe('NavigationSectorMobile', () => {
  let page;
  beforeEach(async () => {
    page = await newSpecPage({
      components: [NavigationSectorMobile],
      html: `<app-navigation-sector-mobile></app-navigation-sector-mobile>`,
    });
  });
  it('should match snapshot', async () => {
    expect(page.root).toMatchSnapshot();
  });
  it('should handle menuItem', async () => {
    const navigation: MenuItem[] = [nav];
    page.root.navigation = navigation;
    page.root.activeSectorId = nav.children[0].id;
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
  it('should handle menuItem w/o children', async () => {
    const { children, ...navWoChildren } = nav;
    const navigation: MenuItem[] = [navWoChildren];
    page.root.navigation = navigation;
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
  it('should handle on keydown esc', async () => {
    page.root.navigation = [nav];
    page.root.hide = jest.fn();
    await page.waitForChanges();
    const anchor = page.root.querySelector('a');
    anchor.dispatchEvent(
      new KeyboardEvent('keydown', {
        view: window,
        bubbles: true,
        cancelable: true,
        key: 'Escape',
      })
    );
    await page.waitForChanges();
    expect(page.root.hide).toHaveBeenCalled();
  });
});
