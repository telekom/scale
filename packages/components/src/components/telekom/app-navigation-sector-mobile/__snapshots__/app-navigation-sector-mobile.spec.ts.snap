// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationSectorMobile should handle menuItem 1`] = `
<app-navigation-sector-mobile>
  <ul class="sector-navigation-mobile">
    <li class="sector-navigation-mobile__item">
      <a aria-current="true" class="sector-navigation-mobile__item-link sector-navigation-mobile__item-link--selected" href="#">
        nav1
        <span class="sr-only">
          active
        </span>
      </a>
    </li>
  </ul>
</app-navigation-sector-mobile>
`;

exports[`NavigationSectorMobile should handle menuItem w/o children 1`] = `
<app-navigation-sector-mobile>
  <ul class="sector-navigation-mobile">
    <li class="sector-navigation-mobile__item">
      <a aria-current="false" class="sector-navigation-mobile__item-link" href="#">
        nav1
      </a>
    </li>
  </ul>
</app-navigation-sector-mobile>
`;

exports[`NavigationSectorMobile should match snapshot 1`] = `
<app-navigation-sector-mobile>
  <ul class="sector-navigation-mobile"></ul>
</app-navigation-sector-mobile>
`;
